### ISSUE TITLE
Boost sanity check logic error

### ISSUE THREAD

**Issue by jpecar** [easybuilders/easybuild-easyblocks#1465]
Around line 254 there's this piece of code:
``` 
if int(pymajorver) >= 3:
    suffix = pymajorver
elif LooseVersion(self.version) >= LooseVersion("1.67.0"):
    suffix = '%s%s' % (pymajorver, pyminorver)
else:
    suffix = ''
custom_paths["files"].append('lib/libboost_python%s.%s' % (suffix, shlib_ext))
```
If I read it right, it sets suffix to major version if python is 3.x. I'm trying to build Boost 1.67 with python 3 and is failing in sanity check, looking for lib/libboost_python3.so while it should look for lib/libboost_python37.so. I assume this logic works for when python is =< 3 and boost is >= 1.67 but not when python is >= 3.

Changing elif to if covers my case but might fail for other cases.

**Comment by ocaisa** [comment]
So (to double-check), would this logic solve the problem:
```
if int(pymajorver) >= 3 and LooseVersion(self.version) < LooseVersion("1.67.0"):
    suffix = pymajorver
elif LooseVersion(self.version) >= LooseVersion("1.67.0"):
    suffix = '%s%s' % (pymajorver, pyminorver)
else:
    suffix = ''
custom_paths["files"].append('lib/libboost_python%s.%s' % (suffix, shlib_ext))
```

**Comment by jhein32** [comment]
Just stumbled over this.  PR #1472 should address the issue.

**Comment by boegel** [comment]
fixed in #1472 which will be included in the upcoming EasyBuild v3.6.3
